coderd/users.go

```diff
@@ -857,8 +857,61 @@ func (api *API) putUserStatus(status database.UserStatus) func(rw http.ResponseW
 			})
 			return
 		}
-		aReq.New = suspendedUser
+		aReq.New = targetUser
+
+		// Notify about the change of user status
+		var key string
+		var templateID uuid.UUID
+		switch status {
+		case database.UserStatusSuspended:
+			key = "suspended_account_name"
+			templateID = notifications.TemplateUserAccountSuspended
+		case database.UserStatusActive:
+			key = "activated_account_name"
+			templateID = notifications.TemplateUserAccountActivated
+		default:
+			api.Logger.Error(ctx, "unable to notify admins as the user's status is unsupported", slog.F("username", user.Username), slog.F("user_status", string(status)))
 
+			httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
+				Message: "Internal error preparing notifications",
+			})
+		}
+
+		// Fetch all users with user admin permissions
+		owners, err := api.Database.GetUsers(ctx, database.GetUsersParams{
+			RbacRole: []string{codersdk.RoleOwner},
+		})
+		if err != nil {
+			httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
+				Message: "Internal error fetching owners",
+				Detail:  err.Error(),
+			})
+			return
+		}
+		userAdmins, err := api.Database.GetUsers(ctx, database.GetUsersParams{
+			RbacRole: []string{codersdk.RoleUserAdmin},
+		})
+		if err != nil {
+			httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
+				Message: "Internal error fetching user admins",
+				Detail:  err.Error(),
+			})
+			return
+		}
+
+		// Send notifications to user admins and affected user
```

Comment:
If we're addressing the affected user, it'll be a bit weird to receive a notification referring to their account in the 3rd person.

`User "bobby" has been suspended` vs `Your account has been suspended`.

There will be different actions to be taken by the different personas, as well.
I think we need to create a new template for this case.

Reply:
Fixed